pwndbg/heap/ptmalloc.py

```diff
@@ -5,17 +5,244 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+from collections import OrderedDict
+
 import gdb
 
 import pwndbg.events
 import pwndbg.typeinfo
+from pwndbg.color import bold
+from pwndbg.color import red
+from pwndbg.constants import ptmalloc
+
+class Heap(pwndbg.heap.heap.BaseHeap):
+    def __init__(self):
+        # Global ptmalloc objects
+        self._main_arena    = None
+        self._mp            = None
+
+        # Symbols and types
+        self._malloc_chunk  = None
+        self._malloc_state  = None
+        self._mallinfo      = None
+        self._malloc_par    = None
+
+
+    @property
+    def main_arena(self):
+        main_arena_addr = pwndbg.symbol.address('main_arena')
+
+        if main_arena_addr is not None:
+            self._main_arena = pwndbg.memory.poi(self.malloc_state, main_arena_addr)
+        else:
+            print(bold(red('Symbol \'main arena\' not found. Try installing libc '
+                           'debugging symbols and try again.')))
+
+        return self._main_arena
+
+
+    @property
+    def mp(self):
+        mp_addr = pwndbg.symbol.address('mp_')
+
+        if mp_addr is not None:
+            self._mp = pwndbg.memory.poi(self.malloc_par, mp_addr)
+
+        return self._mp
+
+
+    @property
+    def malloc_chunk(self):
+        if not self._malloc_chunk:
+            self._malloc_chunk = pwndbg.typeinfo.load('struct malloc_chunk')
```

Comment:
This can't realistically be cached, e.g. if I change the binary I'm debugging without killing GDB, this will be wrong.  I'd recommend just always returning the `typeinfo.load` object, and perhaps using `memoize.reset_on_objfile` to hold the value as cached ~= most of the time.